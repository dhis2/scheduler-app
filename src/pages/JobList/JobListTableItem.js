import React from 'react'
import { bool, shape, string, number } from '@dhis2/prop-types'
import { TableRow, TableCell } from '@dhis2/ui'
import { jobTypesMap } from '../../services/server-translations'
import { ToggleJobSwitch } from '../../components/Switches'
import JobListActions from './JobListActions'
import JobStatus from './JobStatus'
import JobNextRun from './JobNextRun'
import JobSchedule from './JobSchedule'

const JobListTableItem = ({
    job: {
        id,
        displayName,
        jobType,
        cronExpression,
        delay,
        jobStatus,
        nextExecutionTime,
        schedulingType,
        enabled,
    },
}) => (
    <TableRow>
        <TableCell>{displayName}</TableCell>
        <TableCell>{jobTypesMap[jobType]}</TableCell>
        <TableCell>
            <JobSchedule
                cronExpression={cronExpression}
                delay={delay}
                schedulingType={schedulingType}
            />
        </TableCell>
        <TableCell>
            <JobNextRun
                nextExecutionTime={nextExecutionTime}
                enabled={enabled}
            />
        </TableCell>
        <TableCell>
            <JobStatus status={jobStatus} />
        </TableCell>
        <TableCell>
            <ToggleJobSwitch id={id} checked={enabled} />
        </TableCell>
        <TableCell>
            <JobListActions id={id} />
        </TableCell>
    </TableRow>
)

JobListTableItem.propTypes = {
    job: shape({
        displayName: string.isRequired,
        enabled: bool.isRequired,
        id: string.isRequired,
        jobStatus: string.isRequired,
        jobType: string.isRequired,
        schedulingType: string.isRequired,
        cronExpression: string,
        delay: number,
        nextExecutionTime: string,
    }).isRequired,
}

export default JobListTableItem
