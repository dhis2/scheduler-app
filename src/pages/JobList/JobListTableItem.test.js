import React from 'react'
import { shallow } from 'enzyme'
import JobListTableItem from './JobListTableItem'

describe('<JobListTableItem>', () => {
    it('renders cron jobs correctly', () => {
        const job = {
            id: '1',
            displayName: 'Name',
            jobType: 'SEND_SCHEDULED_MESSAGE',
            cronExpression: '0 0 * ? * *',
            jobStatus: 'ENABLED',
            nextExecutionTime: '2100-10-10T14:48:00',
            enabled: true,
            schedulingType: 'CRON',
        }
        const wrapper = shallow(<JobListTableItem job={job} />)

        expect(wrapper).toMatchSnapshot()
    })

    it('renders fixed delay jobs correctly', () => {
        const job = {
            id: '1',
            displayName: 'Name',
            jobType: 'CONTINUOUS_ANALYTICS_TABLE',
            jobStatus: 'ENABLED',
            nextExecutionTime: '',
            enabled: true,
            schedulingType: 'FIXED_DELAY',
            delay: 6000,
        }
        const wrapper = shallow(<JobListTableItem job={job} />)

        expect(wrapper).toMatchSnapshot()
    })
})
