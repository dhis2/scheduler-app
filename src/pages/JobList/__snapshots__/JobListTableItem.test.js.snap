// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobListTableItem> renders cron jobs correctly 1`] = `
<TableRow
  dataTest="dhis2-uicore-tablerow"
>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    Name
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    Send scheduled message
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <JobSchedule
      cronExpression="0 0 * ? * *"
      schedulingType="CRON"
    />
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <JobNextRun
      enabled={true}
      nextExecutionTime="2100-10-10T14:48:00"
    />
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <JobStatus
      status="ENABLED"
    />
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <ToggleJobSwitch
      checked={true}
      id="1"
    />
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <JobListActions
      id="1"
    />
  </TableCell>
</TableRow>
`;

exports[`<JobListTableItem> renders fixed delay jobs correctly 1`] = `
<TableRow
  dataTest="dhis2-uicore-tablerow"
>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    Name
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    Continuous analytics table
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <JobSchedule
      delay={6000}
      schedulingType="FIXED_DELAY"
    />
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <JobNextRun
      enabled={true}
      nextExecutionTime=""
    />
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <JobStatus
      status="ENABLED"
    />
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <ToggleJobSwitch
      checked={true}
      id="1"
    />
  </TableCell>
  <TableCell
    dataTest="dhis2-uicore-tablecell"
  >
    <JobListActions
      id="1"
    />
  </TableCell>
</TableRow>
`;
